staging/src/k8s.io/client-go/rest/fake/fake.go

```diff
@@ -87,23 +89,27 @@ func (c *RESTClient) GetRateLimiter() flowcontrol.RateLimiter {
 }
 
 func (c *RESTClient) request(verb string) *restclient.Request {
-	config := restclient.ContentConfig{
-		ContentType: runtime.ContentTypeJSON,
-		// TODO this was hardcoded before, but it doesn't look right
-		GroupVersion:         &c.APIRegistry.GroupOrDie("").GroupVersion,
-		NegotiatedSerializer: c.NegotiatedSerializer,
+	config := c.ContentConfig
+	defaultContentConfig := restclient.ContentConfig{}
```

Comment:
This was one way I have found to use extensions.GroupName for a Decoder deep in the callstack of kubectl calls.

The decoder created here: 
https://github.com/kubernetes/kubernetes/pull/40956/files#diff-5a429dbc4e41abe93cb1915987b2d66eL107

I want the internalVersion parameter to the `ns.DecoderToVersion` function contain "extensions" in the Group field. 

I am setting the GroupName to "extensions" from the test function here
https://github.com/kubernetes/kubernetes/pull/40956/files#diff-fff79b71b1ff7d6cfb18f869abbc62d5R314

Without this a Decoder is giving me this error:

```
[vagrant@k8s-env cmd]$ go test . -v -run TestApplyTPR
=== RUN   TestApplyTPR
Path: /thirdpartyresources/test-tpr.unit-test.test.com, Method: GET--- FAIL: TestApplyTPR (0.00s)
	cmd_test.go:50: Error running command (exit code 1): error: error when retrieving current configuration of:
		&{0xc42038a7e0 0xc42029e850  test-tpr.unit-test.test.com ../../../test/fixtures/pkg/kubectl/cmd/apply/third-party-resource.yaml &ThirdPartyResource{ObjectMeta:k8s_io_kubernetes_pkg_api_v1.ObjectMeta{Name:test-tpr.unit-test.test.com,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{kubectl.kubernetes.io/last-applied-configuration: ,},OwnerReferences:[],Finalizers:[],ClusterName:,},Description:A TPR definition used for unit tests,Versions:[{v1}],} &TypeMeta{Kind:,APIVersion:,}  false}
		from server for: "../../../test/fixtures/pkg/kubectl/cmd/apply/third-party-resource.yaml": v1beta1.ThirdPartyResource is not suitable for converting to "__internal"
FAIL
exit status 1
FAIL	k8s.io/kubernetes/pkg/kubectl/cmd	0.034s
```

If you have a better way to use "extensions" I can change this impl.